src/controller.cpp

```diff
@@ -57,27 +68,29 @@ geometry_msgs::msg::TwistStamped Controller::computeVelocityCommands(
   const geometry_msgs::msg::Twist & robot_speed,
   nav2_core::GoalChecker * goal_checker)
 {
+  std::chrono::steady_clock::time_point begin = std::chrono::steady_clock::now();
+
   std::lock_guard<std::mutex> lock(*parameters_handler_->getLock());
   nav_msgs::msg::Path transformed_plan = path_handler_.transformPath(robot_pose);
   geometry_msgs::msg::TwistStamped cmd =
-    optimizer_.evalControl(robot_pose, robot_speed, transformed_plan, goal_checker);
+    optimizer_->evalControl(robot_pose, robot_speed, transformed_plan, goal_checker);
 
-  visualize(robot_pose, robot_speed, std::move(transformed_plan));
+  visualize(std::move(transformed_plan));
 
+  std::chrono::steady_clock::time_point end = std::chrono::steady_clock::now();
+  int duration = std::chrono::duration_cast<std::chrono::milliseconds>(end - begin).count();
+  RCLCPP_INFO(logger_, "Controller execution time: %d [ms]", duration);
```

Comment:
Lets make that a debug, if its included at all. What i typically do is use a preprocessor variable https://github.com/ros-planning/navigation2/blob/main/nav2_smac_planner/src/smac_planner_hybrid.cpp#L24 which you can uncomment if you want these kind of metrics https://github.com/ros-planning/navigation2/blob/main/nav2_smac_planner/src/smac_planner_hybrid.cpp#L356-L371 rather than having them inline in the code itself

Reply:
I'll remove it